fdbserver/storageserver.actor.cpp

```diff
@@ -5514,13 +5704,64 @@ ACTOR Future<Void> updateStorage(StorageServer* data) {
 
 		debug_advanceMinCommittedVersion(data->thisServerID, newOldestVersion);
 
+		if (requireCheckpoint) {
+			ASSERT(newOldestVersion == data->pendingCheckpoints.begin()->first);
+			state int idx = 0;
+			for (; idx < data->pendingCheckpoints.begin()->second.size(); ++idx) {
```

Comment:
I wonder if you can refactor the body of this for loop into an ACTOR function, then it's even possible to parallelize multiple checkpoints. Mainly I think `updateStorage` is already long and complicated and better to make it simpler.

Reply:
That's a good idea. Change applied.